README.md

```diff
@@ -3112,6 +3112,24 @@ resource cleanup when possible.
   hash.value?(value)
   ```
 
+* <a name="hash-each"></a>
+  Use `Hash#each_key` instead of `Hash#key.each` and `Hash#each_value`
+  instead of `Hash#values.each` if Hash is not modified in the block.
+<sup>[[link](#hash-each)]</sup>
+
+  ```Ruby
+  # bad
+  hash.keys.each { |k| p k }
+  hash.values.each { |v| p v }
+  hash.each { |k, _v| p k }
+  hash.each { |_k, v| p v }
+
+  # good
+  hash.each_key { |k| p k }
+  hash.each_value { |v| p v }
+  hash.keys.each { |k| hash.delete(k) }
```

Comment:
[Do not modify a collection while traversing it.](https://github.com/bbatsov/ruby-style-guide#no-modifying-collections)